### DISCUSSION TITLE
Accessing ChildContent of FluentDataGrid From Another Component via RenderFragment

### DISCUSSION THREAD

**Discussion by AathifMahir** [microsoft/fluentui-blazor#3109] [Q&A] [answered]
Hello Folks,

I came across the Workload where i need to Wrap a FluentDataGrid into Custom Component Which is Parent Of All where it does provide different features with satisfy the workload but the problem i was facing is Templating the FluentDataGrid from this Custom Component, Where i need to template the Fluent Data Grid from Custom Components RenderFragment Where i want Template Using TemplatedColumn, PropertyColumn but Unfortunely this wasn't possible Since we are not CascadingTheType and Value like Original FluentDataGrid Does Which Passes InternalGridContext to Achieve the Cascading, Therefore i'm looking solution on accessing internalGridContext or DifferentSolution to Achive this Desired Results.

Really Appriciate, if you'll can provide guidence on this, Thanks

**Comment by vnbaaij** [comment]
The internalgridcontext is what the name says: internal. It is not, and will, not be made available.

Not sure what you mean with 'like original FluentDataGrid does'

It is not really clear to me what you are trying to achieve but as a suggestion I'd point you to the Aspire Dashboard source code to see how they handled using the DataGrid.

**Comment by AathifMahir** [comment]
> The internalgridcontext is what the name says: internal. It is not, and will, not be made available.
> 
> Not sure what you mean with 'like original FluentDataGrid does'
> 
> It is not really clear to me what you are trying to achieve but as a suggestion I'd point you to the Aspire Dashboard source code to see how they handled using the DataGrid. 

Let me rephrase the question,

Let's say I have Component Called **CustomDataGrid.razor** which wraps FluentDataGrid

```html
@typeParam TGridItem

<FluentDataGrid Items="@Items">
@ChildContent
</FluentDataGrid>

@code
{
 [Parameter]
 Public IQueryable<TGridItem> Items { get; set; }

[Parameter]
Public RenderFragment? ChildContent { get; set; }
}



```

Now, Let's Use this Component

```html

<CustomDataGrid Items="@MyItems">
  <TemplatedColumn Title="Name">
   <FluentLabel>@context.Name</FluentLabel>
  </TemplatedColumn>

  <PropertyColumn Property="@(p => p.BirthDate)" Format="yyyy-MM-dd" Sortable="true" /> 
......
</CustomDataGrid>
```

Can we achieve this?

**Reply by vnbaaij** [reply]
Have you tried it?

FluentDataGrid is using ChildContent as a way of collecting the list of columns. It might be that that logic interferes with what you want to do

**Reply by AathifMahir** [reply]
> Have you tried it?
> 
> FluentDataGrid is using ChildContent as a way of collecting the list of columns. It might be that that logic interferes with what you want to do 

Yes, I have tried it doesn't seem to work and I'll share the error shortly

**Reply by AathifMahir** [reply]
> Have you tried it?
> 
> FluentDataGrid is using ChildContent as a way of collecting the list of columns. It might be that that logic interferes with what you want to do

Here's the error

![image](https://github.com/user-attachments/assets/6cccf014-c048-45e6-a1d7-d1c02db689ea)

**Reply by AathifMahir** [reply, marked as answer]
> > Have you tried it?
> > FluentDataGrid is using ChildContent as a way of collecting the list of columns. It might be that that logic interferes with what you want to do
> 
> Here's the error
> 
> ![image](https://private-user-images.githubusercontent.com/17165402/399172225-6cccf014-c048-45e6-a1d7-d1c02db689ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU0ODgyNDQsIm5iZiI6MTczNTQ4Nzk0NCwicGF0aCI6Ii8xNzE2NTQwMi8zOTkxNzIyMjUtNmNjY2YwMTQtYzA0OC00NWU2LWExZDctZDFjMDJkYjY4OWVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI5VDE1NTkwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZlODBhNTM4N2ExOTZlZDhkNThlZDA5OTc4MmMyYzlhZGIxNmQ1YTBjNDc4YjA1ZTk0ZDhmZTkyNmM2MjllMGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.PSyJjsWTNmrb9yvipw6uxmFvvbDm4oqWSm0VSgUGeiw)

Nevermind, I have managed to fix the issue by Adding CascadingTypeParameter as An Attribute

````html
@typeParam TGridItem
@attribute [CascadingTypeParameter(nameof(TGridItem))]
````
